From 2138af6b7d0c22e9ea48362ffb54d1bcbc1e33a7 Mon Sep 17 00:00:00 2001
Date: Fri, 9 Feb 2024 19:29:07 +0100
Subject: [PATCH] Exclude IOKit-specifics from QGlobal when targeting iOS

---
 src/corelib/global/qglobal.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/corelib/global/qglobal.cpp b/src/corelib/global/qglobal.cpp
index 01f534cc..1abf6a27 100644
--- a/src/corelib/global/qglobal.cpp
+++ b/src/corelib/global/qglobal.cpp
@@ -88,7 +88,7 @@
 #  include <sys/systeminfo.h>
 #endif
 
-#if defined(Q_OS_DARWIN) && QT_HAS_INCLUDE(<IOKit/IOKitLib.h>)
+#if defined(Q_OS_DARWIN) && !defined(Q_OS_IOS) && QT_HAS_INCLUDE(<IOKit/IOKitLib.h>)
 #  include <IOKit/IOKitLib.h>
 #  include <private/qcore_mac_p.h>
 #endif
@@ -2912,7 +2912,7 @@ enum {
 */
 QByteArray QSysInfo::machineUniqueId()
 {
-#if defined(Q_OS_DARWIN) && QT_HAS_INCLUDE(<IOKit/IOKitLib.h>)
+#if defined(Q_OS_DARWIN) && !defined(Q_OS_IOS) && QT_HAS_INCLUDE(<IOKit/IOKitLib.h>)
     char uuid[UuidStringLen + 1];
     io_service_t service = IOServiceGetMatchingService(kIOMasterPortDefault, IOServiceMatching("IOPlatformExpertDevice"));
     QCFString stringRef = (CFStringRef)IORegistryEntryCreateCFProperty(service, CFSTR(kIOPlatformUUIDKey), kCFAllocatorDefault, 0);
-- 
2.43.0

