diff --git a/src/portmidi/mac/pmmacosxcm.c b/src/portmidi/mac/pmmacosxcm.c
index eb37c72..9932fd1 100755
--- a/src/portmidi/mac/pmmacosxcm.c
+++ b/src/portmidi/mac/pmmacosxcm.c
@@ -814,7 +814,7 @@ static CFStringRef ConnectedEndpointName(MIDIEndpointRef endpoint)
     if (nConnected) {
       const SInt32 *pid = (const SInt32 *)(CFDataGetBytePtr(connections));
       for (i = 0; i < nConnected; ++i, ++pid) {
-        MIDIUniqueID id = EndianS32_BtoN(*pid);
+        MIDIUniqueID id = CFSwapInt32BigToHost(*pid);
         MIDIObjectRef connObject;
         MIDIObjectType connObjectType;
         err = MIDIObjectFindByUniqueID(id, &connObject, &connObjectType);
-- 
2.35.1

