diff --git a/mkspecs/features/uikit/xcodebuild.mk b/mkspecs/features/uikit/xcodebuild.mk
index 0c8d99f4..7c45d68e 100644
--- a/mkspecs/features/uikit/xcodebuild.mk
+++ b/mkspecs/features/uikit/xcodebuild.mk
@@ -71,11 +71,11 @@ endif
 %-simulator: DEVICES = $(firstword $(SIMULATOR_DEVICES))
 %-device: DEVICES = $(HARDWARE_DEVICES)
 
-GENERIC_DEVICE_DESTINATION := $(EXPORT_GENERIC_DEVICE_DESTINATION)
-GENERIC_SIMULATOR_DESTINATION := "id=$(shell $(MAKEFILE_DIR)devices.py $(EXPORT_DEVICE_FILTER) | tail -n 1)"
+GENERIC_DEVICE_DESTINATION := "$(EXPORT_GENERIC_DEVICE_DESTINATION)"
+GENERIC_SIMULATOR_DESTINATION := "$(EXPORT_GENERIC_SIMULATOR_DESTINATION)"
 
-%-simulator: DESTINATION = $(if $(DESTINATION_ID),"id=$(DESTINATION_ID)",$(GENERIC_SIMULATOR_DESTINATION))
-%-device: DESTINATION = $(if $(DESTINATION_ID),"id=$(DESTINATION_ID)",$(GENERIC_DEVICE_DESTINATION))
+%-simulator: DESTINATION = $(GENERIC_SIMULATOR_DESTINATION)
+%-device: DESTINATION = $(GENERIC_DEVICE_DESTINATION)
 
 XCODE_VERSION_MAJOR := $(shell xcodebuild -version | grep Xcode | sed -e 's/Xcode //' | sed -e 's/\..*//')
 
diff --git a/mkspecs/features/uikit/xcodebuild.prf b/mkspecs/features/uikit/xcodebuild.prf
index 7a6b2acf..7ea51362 100644
--- a/mkspecs/features/uikit/xcodebuild.prf
+++ b/mkspecs/features/uikit/xcodebuild.prf
@@ -42,6 +42,7 @@ SIMULATOR_SDK = $${simulator.sdk}
 ios {
     DEVICE_FILTER = --platform iOS --minimum-deployment-target $$QMAKE_IOS_DEPLOYMENT_TARGET
     GENERIC_DEVICE_DESTINATION = "generic/platform=iOS"
+    GENERIC_SIMULATOR_DESTINATION = "generic/platform=iOS Simulator"
 }
 tvos {
     DEVICE_FILTER = --platform tvOS --minimum-deployment-target $$QMAKE_TVOS_DEPLOYMENT_TARGET
@@ -51,7 +52,7 @@ watchos {
     DEVICE_FILTER = --platform watchOS --minimum-deployment-target $$QMAKE_WATCHOS_DEPLOYMENT_TARGET
     GENERIC_DEVICE_DESTINATION = "generic/platform=watchOS"
 }
-QMAKE_EXTRA_VARIABLES += DEVICE_SDK SIMULATOR_SDK DEVICE_FILTER GENERIC_DEVICE_DESTINATION
+QMAKE_EXTRA_VARIABLES += DEVICE_SDK SIMULATOR_SDK DEVICE_FILTER GENERIC_DEVICE_DESTINATION GENERIC_SIMULATOR_DESTINATION
 
 QMAKE_EXTRA_INCLUDES += $$shell_quote($$PWD/xcodebuild.mk)
 
-- 
2.35.1

