From 210e90b9a1c5438a7dae2b4e2e6e111b7cf45263 Mon Sep 17 00:00:00 2001
Date: Sun, 3 Mar 2024 00:22:33 +0100
Subject: [PATCH] Include public WASM toolchain helpers

---
 cmake/QtWasmHelpers.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/QtWasmHelpers.cmake b/cmake/QtWasmHelpers.cmake
index 614a677a..eff8d7de 100644
--- a/cmake/QtWasmHelpers.cmake
+++ b/cmake/QtWasmHelpers.cmake
@@ -1,6 +1,7 @@
 # Copyright (C) 2022 The Qt Company Ltd.
 # SPDX-License-Identifier: BSD-3-Clause
 
+include("${CMAKE_CURRENT_LIST_DIR}/QtPublicWasmToolchainHelpers.cmake")
 
 # WARNING must keep in sync with wasm-emscripten/qmake.conf!
 function (qt_internal_setup_wasm_target_properties wasmTarget)
-- 
2.43.0

